enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -392,12 +392,13 @@ func (s *COWStore) copyChunkIfNotDirty(chunkStartOffset int64) (err error) {
 	}
 
 	dstChunkSize := s.calculateChunkSize(chunkStartOffset)
+
+	s.mu.Lock()
 	dst, err := s.initDirtyChunk(chunkStartOffset, dstChunkSize)
 	if err != nil {
+		s.mu.Unlock()
 		return status.WrapError(err, "initialize dirty chunk")
 	}
-
-	s.mu.Lock()
 	src := s.chunks[chunkStartOffset]
 	s.chunks[chunkStartOffset] = dst
 	s.dirty[chunkStartOffset] = true
```

Comment:
If we move this logic into `initDirtyChunk` and have that func return `src, dst, err`, then we can make this a bit safer by doing the `Lock() / defer Unlock()` at the top of `initDirtyChunk` and removing the locking here.